hashring/hashring.go

```diff
@@ -41,16 +42,17 @@ type HashRingConfiguration struct {
 	ReplicaPoints int
 }
 
+// HashRing stores strings on a consistent hash ring. HashRing internally uses
```

Comment:
I think an alternative to the approach of making exported fns thread-safe and non-exported not would be to have two types: `HashRing` and `ConcurrentHashRing` both with the safe interface. `ConcurrentHashRing` would be composed of a `HashRing` and simply delegate fn calls to it. I think that type of approach is closer to the Java approach.